modules/custom/activity_viewer/src/Plugin/views/row/ActivityRow.php

```diff
@@ -30,8 +31,39 @@ public function preRender($result) {
 
       foreach ($result as $row) {
         $render_result = [];
-        $render_result[] = $row;
         $entity = $row->_entity;
+        $target_type = $entity->get('field_activity_entity')->target_type;
+
+        if (!empty($target_type) && $target_type === "comment") {
+          $comment_id = $entity->get('field_activity_entity')->target_id;
+          $comment_storage = \Drupal::entityTypeManager()->getStorage('comment');
+          $comment = $comment_storage->load($comment_id);
+          $parent_id = $comment->get('entity_id')->target_id;
+          $parent_type = $comment->get('entity_type')->value;
+
+          if (!empty($parent_id) && !empty($parent_type)) {
+            if ($parent_type === "post") {
+              $parent_storage = \Drupal::entityTypeManager()->getStorage($parent_type);
+              $parent_entity = $parent_storage->load($parent_id);
+              $current_user_id = \Drupal::currentUser()->id();
+              $current_user = User::load($current_user_id);
+
+              $parent_status = $parent_entity->get('status')->value;
+              $current_user_is_admin = $current_user->hasPermission('view unpublished post entities');
+
+              if ((!empty($parent_status) && $parent_status !== "0")
+                || $current_user_is_admin) {
+                $render_result[] = $row;
+              }
+            }
+            else {
```

Comment:
You don't need else statements usually. 
What you could do instead of all the elses is just populate the `$row` variable.

Reply:
true, I will do that